.github/workflows/python-ci.yml

```diff
@@ -21,39 +21,59 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
+        runs-on: [ubuntu-latest]
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
+        flint-inst-cmd: ["sudo apt-get install libflint-dev"]
         include:
-          - runs-on: ubuntu-latest
-            flint-inst-cmd: sudo apt-get install libflint-dev
           - runs-on: macos-latest
             flint-inst-cmd: brew install flint
+            python-version: 3.8
+          - runs-on: macos-latest
+            flint-inst-cmd: brew install flint
+            python-version: 3.11
     #          - runs-on: windows-latest
     #            flint-inst-cmd:
     steps:
       - uses: actions/checkout@v3
         with:
           submodules: recursive
           fetch-depth: 0
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          ${{ matrix.flint-inst-cmd }}
+      - name: Setup nox
+        run: pipx install nox
+      - name: Test on 🐍 ${{ matrix.python-version }}
+        run: nox -s tests-${{ matrix.python-version }}
+  coverage:
+    name: 🐍 ${{ matrix.python-version }} Coverage
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.8", "3.11"]
+        flint-inst-cmd: ["sudo apt-get install libflint-dev"]
```

Comment:
```suggestion
```
This can be inlined as coverage is only collected on Linux and the command is the same for everybody here.